### ISSUE TITLE
Divide by zero if unclip_ratio is changed

### ISSUE THREAD

**Issue by nh2** [mindee/doctr#1375]
### 🚀 The feature

Hi,

unclear to me if this is a bug or some clever dependency between hardcoded constants, so filing it as a "feature request" that changing these constants should work:

I have some inputs where a larger unclip ratio seems to help improve results:

```diff
-        self.unclip_ratio = 1.5 if assume_straight_pages else 2.2
+        self.unclip_ratio = 3 if assume_straight_pages else 5
```

in

https://github.com/mindee/doctr/blob/75bddfce9f2c37e6f07c0acd9b00c50ba77ce763/doctr/models/detection/differentiable_binarization/base.py#L40

But if I do that, in some rare cases, this code causes a `divide by zero`:

https://github.com/mindee/doctr/blob/75bddfce9f2c37e6f07c0acd9b00c50ba77ce763/doctr/utils/geometry.py#L277-L286

In here, it's `xright - xleft` being `[0., 0.]`.

The code continues, with `estimate_page_angle()` returning `0.0`, but I fear that due to the calling code

https://github.com/mindee/doctr/blob/75bddfce9f2c37e6f07c0acd9b00c50ba77ce763/doctr/models/builder.py#L56-L61

something might get incorrectly rotated in the end.

---

So I'm wondering:

* Should the code be robust to allow changes in `unclip_ratio`?
* What guarantees that the current `unclip_ratio` avoids this issue, or are we just lucky that it happens less often?

Thanks!

### Motivation, pitch

It would be nice to be able to change `unclip_ratio` safely.

**Comment by felixT2K** [comment]
Hi @nh2 thanks for opening this i will have a look :)

For the moment instead of tweaking around with the `unclip_ratio` you can configure the `bin_thresh` value for the detection model (higher value = more precise & possible less detections  | lower value = less precise & more detections)

example:

```
det_model = db_resnet50(pretrained=True, bin_thresh=0.7)

predictor = ocr_predictor(det_arch=det_model, reco_arch="parseq", pretrained=True)
```

Best regards,
Felix

**Comment by felixdittrich92** [comment]
TODO: @felixdittrich92 Fix possible divide by zero

**Comment by felixdittrich92** [comment]
Hi @nh2 :wave:,

I have applied a short fix to avoid this runtime warning.
But instead of tweaking around with the `unclip_ratio` you should try before to modify the binarization threshold and/or box threshold :)
https://mindee.github.io/doctr/latest/using_doctr/using_models.html#advanced-options

**Comment by nh2** [comment]
@felixdittrich92 Thanks!

The reason we increased `unclip_ratio` was this:

According to my understanding, doctr first finds text, fitting a suitable box around it, grows the box by `unclip_ratio`, and then runs detection on that box to extract the text.

We found that the boxes are sometimes not grown enough for text extraction to work well.

Would changing `box_thresh` have the same effect?

From

https://github.com/mindee/doctr/blob/dd1fbbe69903321f30a86aec264b491038d57a30/doctr/models/detection/core.py#L21-L22

it sounds they are a bit different.
